src/utils/streams.js

```diff
@@ -0,0 +1,304 @@
+  writeStream.write(JSON.stringify(jsonArray[0]));
+  jsonArray.shift();
+  jsonArray.forEach((line) => {
+    writeStream.write(',\n  ');
+    writeStream.write(JSON.stringify(line));
+  });
+  writeStream.write('\n]\n');
+  writeStream.end();
+};
+
+const _convertFileFun = (fileName, writeStream, help) => {
+  csv()
+    .fromFile(fileName)
+    .then((jsonArray) => {
+      if (jsonArray.length === 0) {
+        writeStream.write('\n');
+        writeStream.end();
+      } else if (jsonArray.length === 1) {
+        writeStream.write(`${JSON.stringify(jsonArray[0])}\n`);
+        writeStream.end();
+      } else {
+        _outputFormattedMultilineJson(jsonArray, writeStream);
+      }
+    })
+    .catch((err) => {
+      console.log(err.message);
+      console.log(help);
+    });
+};
+
+const _getCssFilesForPath = (pathToCss) => {
+  try {
+    let files = fs.readdirSync(pathToCss);
+    files = files
+      .filter((file) => path.extname(file) === '.css')
+      .filter((file) => file !== 'bundle.css')
+      .map((file) => path.join(pathToCss, file));
+
+    if (files.length === 0) {
+      console.log('ERROR. There are no css files in the given directory.');
+      return [];
+    }
+
+    return files;
+  } catch (err) {
+    console.log(err.message);
+    return [];
+  }
+};
+
+const _removeFile = (fileName) => {
+  try {
+    fs.unlinkSync(fileName);
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      // Do nothing.
+    } else {
+      console.log(err.message);
+      return false;
+    }
+  }
+
+  return true;
+};
+
+const _writeAllFilesToBundleCss = (fileArray, bundleCssFile, help) => {
+  const bundleCssStream = fs.createWriteStream(bundleCssFile, { flags: 'a' });
+  const currentCssFile = fileArray.shift();
+  bundleCssStream.on('finish', () => {
+    if (fileArray.length > 0) {
+      _writeAllFilesToBundleCss(fileArray, bundleCssFile, help);
+    }
+  });
+
+  const readStream = fs.createReadStream(currentCssFile);
+  readStream.on('error', (err) => {
+    console.log(err.message);
+    console.log(help);
+  });
+
+  readStream
+    .pipe(bundleCssStream);
+};
+
+/**
+ * Reverts each line from stdin.
+ */
+const reverseFunc = () => {
+  _processStdin(_reverseString);
+};
+
+/**
+ * Converts each line from stdin to Upper case.
+ */
+const transformFunc = () => {
+  _processStdin(_convertStringToUpperCase);
+};
+
+/**
+ * Concats all css files in the provided dir to the bundle.css file.
+ */
+const cssBundleFunc = () => {
+  const help = readmeObject.cssBundle.join('\n');
+  const pathToCss = argv.path;
+
+  if (!_validateStringParam(pathToCss, `${PATH_IS_INVALID}\n${help}`)) {
+    return;
+  }
+
+  const files = _getCssFilesForPath(pathToCss);
+  if (files.length === 0) {
+    console.log(help);
+    return;
+  }
+
+  const bundleCssFile = path.join(pathToCss, 'bundle.css');
+  if (!_removeFile(bundleCssFile)) {
+    console.log(help);
+    return;
+  }
+
+  files.push(path.join(__dirname, 'nodejs18-hw3-css.css'));
```

Comment:
I believe that the whole meaning of the requirement below:

> Add contents of ​https://epa.ms/nodejs18-hw3-css​​​​to the end of the result file

was to build css bundle from different sources (not only files but yet remote resource). if we download the resource and put it into file system the requirements looses any sense. Could you request this file via http instead ?

Reply:
This link ends on epam's onedrive personal page after few redirects.
Don't think it's a good idea to reveal this link in the public repo.
Also getting to this link requires two factor authentication. 

I've updated code the way when it's easy to switch getting data from a file to getting data from a url.